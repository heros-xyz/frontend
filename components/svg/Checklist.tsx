import { Icon, IconProps } from "@chakra-ui/react";
import React from "react";

export const IconPage: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 32 32"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.4993 16H18.4993M13.4993 19.3333H18.4993M20.166 23.5H11.8327C10.9122 23.5 10.166 22.7538 10.166 21.8333V10.1667C10.166 9.24619 10.9122 8.5 11.8327 8.5H16.4875C16.7085 8.5 16.9205 8.5878 17.0768 8.74408L21.5886 13.2559C21.7449 13.4122 21.8327 13.6242 21.8327 13.8452V21.8333C21.8327 22.7538 21.0865 23.5 20.166 23.5Z"
        stroke="currentColor"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <rect
        x="0.5"
        y="0.5"
        width="31"
        height="31"
        rx="5.5"
        stroke="currentColor"
      />
    </Icon>
  );
};

export const IconCheck: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 33 34"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect y="0.5" width="33" height="33" rx="16.5" fill="currentColor" />
      <path
        d="M10 17.5L14 21.5L24 11.5"
        stroke="white"
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Icon>
  );
};

export const IconBasic: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 32 32"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.3327 11.8333C19.3327 13.6743 17.8403 15.1667 15.9993 15.1667C14.1584 15.1667 12.666 13.6743 12.666 11.8333C12.666 9.99238 14.1584 8.5 15.9993 8.5C17.8403 8.5 19.3327 9.99238 19.3327 11.8333Z"
        stroke="currentColor"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M15.9993 17.6667C12.7777 17.6667 10.166 20.2783 10.166 23.5H21.8327C21.8327 20.2783 19.221 17.6667 15.9993 17.6667Z"
        stroke="currentColor"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <rect
        x="0.5"
        y="0.5"
        width="31"
        height="31"
        rx="5.5"
        stroke="currentColor"
      />
    </Icon>
  );
};

export const IconSport: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 32 32"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21.3033 21.3033C24.2322 18.3744 24.2322 13.6256 21.3033 10.6967C18.3744 7.76777 13.6256 7.76777 10.6967 10.6967M21.3033 21.3033C18.3744 24.2322 13.6256 24.2322 10.6967 21.3033C7.76777 18.3744 7.76777 13.6256 10.6967 10.6967M21.3033 21.3033L16 16L10.6967 10.6967"
        stroke="currentColor"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M9.33398 16.833C12.5556 16.833 15.1673 19.4447 15.1673 22.6663"
        stroke="currentColor"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M23.5 15.167C20.2783 15.167 17.6667 12.5553 17.6667 9.33366"
        stroke="currentColor"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <rect
        x="0.5"
        y="0.5"
        width="31"
        height="31"
        rx="5.5"
        stroke="currentColor"
      />
    </Icon>
  );
};

export const IconCarrer: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 32 32"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.5 23.5V20.1667M8.5 20.1667V10.1667C8.5 9.24619 9.24619 8.5 10.1667 8.5H15.5833L16.4167 9.33333H23.5L21 14.3333L23.5 19.3333H16.4167L15.5833 18.5H10.1667C9.24619 18.5 8.5 19.2462 8.5 20.1667ZM16 8.91667V13.5"
        stroke="currentColor"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <rect
        x="0.5"
        y="0.5"
        width="31"
        height="31"
        rx="5.5"
        stroke="currentColor"
      />
    </Icon>
  );
};
