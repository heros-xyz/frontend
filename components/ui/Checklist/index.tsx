import { Box, Flex, Heading, Text, Link } from "@chakra-ui/react";
import React from "react";
import { If, Then, Else } from "react-if";
import NextLink from "next/link";
import {
  IconCheck,
  IconPage,
  IconBasic,
  IconSport,
  IconCarrer,
} from "@/components/svg/Checklist";

export interface ChecklistProps {
  title: string;
  description: string;
  checked: boolean;
  type: "page" | "basic" | "sport" | "career";
  responseType:
    | "hasBasicInformation"
    | "hasPageInformation"
    | "hasSportProfile"
    | "hasCareerJourney";
  link?:
    | "basic-information"
    | "career-journey"
    | "page-information"
    | "sport-profile";
}

const Checklist: React.FC<ChecklistProps> = ({
  title,
  description,
  checked,
  type,
  link,
}) => {
  const Icon = () => {
    switch (type) {
      case "page":
        return (
          <IconPage
            color={checked ? "primary" : "white"}
            w={{ base: "32px", lg: "48px" }}
            h={{ base: "32px", lg: "48px" }}
          />
        );
      case "basic":
        return (
          <IconBasic
            color={checked ? "primary" : "white"}
            w={{ base: "32px", lg: "48px" }}
            h={{ base: "32px", lg: "48px" }}
          />
        );
      case "sport":
        return (
          <IconSport
            color={checked ? "primary" : "white"}
            w={{ base: "32px", lg: "48px" }}
            h={{ base: "32px", lg: "48px" }}
          />
        );
      case "career":
        return (
          <IconCarrer
            color={checked ? "primary" : "white"}
            w={{ base: "32px", lg: "48px" }}
            h={{ base: "32px", lg: "48px" }}
          />
        );
      default:
        break;
    }
  };
  return (
    <Box
      bg={checked ? "secondary" : "primary"}
      p={{ base: "14px", lg: "30px" }}
      borderRadius="md"
      border={checked ? "2px" : ""}
      borderStyle={checked ? "solid" : ""}
      borderColor={checked ? "primary" : ""}
      h={checked ? { lg: "228px" } : { lg: "312px" }}
      position={{ lg: "relative" }}
    >
      <Flex
        gap={3}
        direction={{ base: "row", lg: "column" }}
        justify="space-between"
        align={{ base: "center", lg: "flex-start" }}
      >
        <Box>{Icon()}</Box>
        <Box flex={1}>
          <Heading
            color={checked ? "primary" : "white"}
            fontSize={{ base: "sm", lg: "xl" }}
            fontWeight="bold"
            lineHeight={{ base: "20px", lg: "28px" }}
            textTransform="capitalize"
            marginBottom={0.5}
          >
            {title}
          </Heading>
          <Text
            as="p"
            color={checked ? "primary" : "white"}
            fontSize={{ base: "10px", lg: "16px" }}
            letterSpacing={{ base: "small", lg: "normal" }}
          >
            {description}
          </Text>
        </Box>
        <If condition={checked}>
          <Then>
            <Box
              position={{ lg: "absolute" }}
              top={{ lg: "30px" }}
              right={{ lg: "30px" }}
            >
              <IconCheck
                w={{ base: "32px", lg: "48px" }}
                h={{ base: "32px", lg: "48px" }}
              />
            </Box>
          </Then>
          <Else>
            <Link
              as={NextLink}
              href={`/athlete/onboarding/${link}`}
              w={{ lg: "100%" }}
            >
              <Box
                pt={{ base: "8px", lg: "10px" }}
                pb={{ base: "8px", lg: "10px" }}
                pr={{ base: "16px", lg: "20px" }}
                pl={{ base: "16px", lg: "20px" }}
                bg="secondary"
                borderRadius={3}
                w={{ base: "48px", lg: "full" }}
                h={{ base: "28px", lg: "48px" }}
                marginTop={{ lg: "30px" }}
              >
                <Text
                  fontSize={{ base: "xxs", lg: "lg" }}
                  fontWeight="bold"
                  lineHeight={{ base: "12px", lg: "28px" }}
                  display="flex"
                  alignItems="center"
                  textAlign={{ base: "center", lg: "center" }}
                  justifyContent={{ lg: "center" }}
                >
                  GO
                </Text>
              </Box>
            </Link>
          </Else>
        </If>
      </Flex>
    </Box>
  );
};

export default Checklist;
