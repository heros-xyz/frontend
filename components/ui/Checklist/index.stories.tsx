import type { Meta, StoryObj } from "@storybook/react";
import Checklist from ".";

const meta: Meta<typeof Checklist> = {
  title: "Athlete/Checklist",
  component: Checklist,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {
    title: {
      options: [
        "Page Information",
        "Sport Profile",
        "Career Journey",
        "Personal Information",
      ],
      control: { type: "select" },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Checklist>;
export const CareerJourney: Story = {
  render: (args) => <Checklist {...args}>Join with us</Checklist>,
  args: {
    type: "career",
    title: "Career Journey",
    description:
      "Make sure you paint a compelling picture of how they can join you on this journey.",
    checked: false,
    link: "career-journey",
  },
};
export const SportProfile: Story = {
  render: (args) => <Checklist {...args}>Join with us</Checklist>,
  args: {
    type: "sport",
    title: "Sport Profile",
    description:
      "Fan will appreciate more if they know how hard you trained & worked to be the best!",
    checked: false,
    link: "sport-profile",
  },
};
export const BasicInformation: Story = {
  render: (args) => <Checklist {...args}>Join with us</Checklist>,
  args: {
    type: "basic",
    title: "Personal Information",
    description: "Help your fan & followers know you better personally.",
    checked: true,
    link: "basic-information",
  },
};
export const PageInformation: Story = {
  render: (args) => <Checklist {...args}>Join with us</Checklist>,
  args: {
    type: "page",
    title: "Page Information",
    description:
      "Make sure you paint a compelling picture of how they can join you on this journey.",
    checked: true,
    link: "page-information",
  },
};
