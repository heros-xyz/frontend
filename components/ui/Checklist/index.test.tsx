import { render } from "@testing-library/react";
import Checklist from ".";

it("renders homepage unchanged", () => {
  const { container } = render(
    <Checklist
      type="career"
      title="Page Information"
      description="Make sure you paint a compelling picture of how they can join you on this journey."
      checked={true}
      responseType="hasPageInformation"
    />
  );
  expect(container).toMatchSnapshot();
});
