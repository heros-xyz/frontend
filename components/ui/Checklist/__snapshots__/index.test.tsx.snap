// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-1updty1"
  >
    <div
      class="css-iwomsf"
    >
      <div
        class="css-0"
      >
        <svg
          class="chakra-icon css-1tuorrv"
          focusable="false"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.5 23.5V20.1667M8.5 20.1667V10.1667C8.5 9.24619 9.24619 8.5 10.1667 8.5H15.5833L16.4167 9.33333H23.5L21 14.3333L23.5 19.3333H16.4167L15.5833 18.5H10.1667C9.24619 18.5 8.5 19.2462 8.5 20.1667ZM16 8.91667V13.5"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <rect
            height="31"
            rx="5.5"
            stroke="currentColor"
            width="31"
            x="0.5"
            y="0.5"
          />
        </svg>
      </div>
      <div
        class="css-1rr4qq7"
      >
        <h2
          class="chakra-heading css-1x1z8k8"
        >
          Page Information
        </h2>
        <p
          class="chakra-text css-1acxzk7"
        >
          Make sure you paint a compelling picture of how they can join you on this journey.
        </p>
      </div>
      <div
        class="css-fr60pb"
      >
        <svg
          class="chakra-icon css-1tuorrv"
          focusable="false"
          viewBox="0 0 33 34"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="currentColor"
            height="33"
            rx="16.5"
            width="33"
            y="0.5"
          />
          <path
            d="M10 17.5L14 21.5L24 11.5"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
